app/renderer/reducers/contextMenuReducer.js

```diff
@@ -33,6 +41,45 @@ const validateState = function (state) {
   return state
 }
 
+function generateMuteFrameList (framePropsList, muted) {
+  return framePropsList.map((frameProp) => {
+    return {
+      frameKey: frameProp.get('key'),
+      tabId: frameProp.get('tabId'),
+      muted: muted && frameProp.get('audioPlaybackActive') && !frameProp.get('audioMuted')
+    }
+  })
+}
+
+const onTabPageMenu = function (state, action) {
+  action = validateAction(action)
+  state = validateState(state)
+
+  const frames = frameStateUtil.getNonPinnedFrames(state) || Immutable.List()
+  const tabsPerPage = Number(getSetting(settings.TABS_PER_PAGE))
+  const tabPageFrames = frames.slice(action.get('index') * tabsPerPage, (action.get('index') * tabsPerPage) + tabsPerPage)
```

Comment:
Would be good to read `action.get('index')` into a variable and skip processing if:
- index is null /undefined
- index is out of bounds (negative value, over the max number of tab pages)